kong/plugins/session/access.lua

```diff
@@ -52,20 +52,20 @@ function _M.execute(conf)
                                        load_consumer, cid)
 
   if err then
-    ngx.log(ngx.ERR, "Error loading consumer: ", err)
+    kong.log.err("Error loading consumer: ", err)
     return
   end
 
   -- destroy sessions with invalid consumer_id
   if not consumer then
-    ngx.log(ngx.DEBUG, "No consumer, destroying session")
+    kong.log.debug("No consumer, destroying session")
     return s:destroy()
   end
 
   s:start()
 
   set_consumer(consumer, credential)
-  ngx.ctx.authenticated_session = s
+  kong.ctx.authenticated_session = s
```

Comment:
AFAIK it should be `kong.ctx.shared.authenticated_session` or `kong.ctx.plugin.authenticated_session` depending whether the value is used to for this plugin or shared with other plugins.

Reply:
We have had plugins destroy a session based on some plugin chain so changed all to `shared`.